src/api/catalog.js

```diff
@@ -62,7 +62,13 @@ export default ({config, db}) => function (req, res, body) {
 	}
 
 	// pass the request to elasticsearch
-	let url = config.elasticsearch.host + ':' + config.elasticsearch.port + (req.query.request ? _updateQueryStringParameter(req.url, 'request', null) : req.url)
+	let port = config.elasticsearch.port
+	if (!port || port.length < 2) {
+		port = ''
+	} else {
+		port = `:${port}`
+	}
+	let url = `${config.elasticsearch.protocol}://${config.elasticsearch.host}${port}${(req.query.request ? _updateQueryStringParameter(req.url, 'request', null) : req.url)}`
```

Comment:
This snippet is being used here for the third time we need to make it kind of helper please